File: algorithm/20181024/stack.py
class stack:
    array = []

    def __init__(self):
        self.array = []

    def pop(self):
        if self.empty() is False:
            return self.array.pop(self.size()-1)
        else:
            return -1

    def push(self, val):
        return self.array.append(val)

    def top(self):
        if self.empty() is False:
            point = self.size()-1
            return self.array[point]
        else:
            return -1


    def empty(self):
        if self.size() == 0:
            return True
        else:
            return False

    def size(self):
        return len(self.array)

File: JAVA/class/stack.java

import java.util.*;

public class Stack {
	
	ArrayList<Integer> al = null;
	
	public Stack(){
		this.al = new ArrayList<Integer>();
	}
	
	public void push(int data){
		this.al.add(data);
	}
	
	public int size(){
		return this.al.size();
	}
	
	public Boolean empty(){
		if(this.size() > 0){
			return false;
		}
		else{
			return true;
		}
	}
	
	public int pop(){
		if(this.empty() == false){
			int ret_Int = this.al.get(this.al.size()-1);
			this.al.remove(this.al.size()-1);
			return ret_Int; 
		}
		else{
			return -1;
		}
	}
	
	
	public static void main(String[] args) {
		Stack s1 = new Stack();
		s1.push(1);
		s1.push(2);
		s1.push(3);
		
		while(s1.empty() == false){
			System.out.println(s1.pop());
		}

	}

}
